soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -46,76 +167,634 @@ Color_RGBA8 Color_Skipped_Main_Default              = { 160, 160, 160, 255 };
+            ((gSaveContext.n64ddFlag || CVarGetInteger("gGravediggingTourFix", 0)) &&
+                gSaveContext.sceneFlags[scene].collect & (1 << flag) || (gPlayState->actorCtx.flags.collect & (1 << flag))); // rando/fix flag
+    default:
+        return false;
+    }
+    return false;
+}
+
+void CheckTrackerDialogClosed() {
+    if (messageCloseCheck) {
+        messageCloseCheck = false;
+    }
+}
+
+void CheckTrackerShopSlotChange(uint8_t cursorSlot, int16_t basePrice) {
+    if (gPlayState->sceneNum == SCENE_HAPPY_MASK_SHOP) { // Happy Mask Shop is not used in rando, so is not tracked
+        return;
+    }
+
+    auto slot = startingShopItem.find(gPlayState->sceneNum)->second + cursorSlot;
+    auto status = gSaveContext.checkTrackerData[slot].status;
+    if (status == RCSHOW_SEEN) {
+        gSaveContext.checkTrackerData[slot].status = RCSHOW_IDENTIFIED;
+        gSaveContext.checkTrackerData[slot].price = basePrice;
+        SaveManager::Instance->SaveSection(gSaveContext.fileNum, sectionId, true);
+    }
+}
+
+void CheckTrackerTransition(uint32_t sceneNum) {
+    if (!GameInteractor::IsSaveLoaded()) {
+        return;
+    }
+    gSaveContext;
+    if (transitionCheck) {
+        transitionCheck = false;
+    }
+    doAreaScroll = true;
+    previousArea = currentArea;
+    currentArea = GetCheckArea();
+    switch (sceneNum) {
+        case SCENE_KOKIRI_SHOP:
+        case SCENE_BAZAAR:
+        case SCENE_POTION_SHOP_MARKET:
+        case SCENE_BOMBCHU_SHOP:
+        case SCENE_POTION_SHOP_KAKARIKO:
+        case SCENE_GORON_SHOP:
+        case SCENE_ZORA_SHOP:
+            SetShopSeen(sceneNum, false);
+            break;
+    }
+}
+
+void CheckTrackerFrame() {
+    if (!GameInteractor::IsSaveLoaded()) {
+        return;
+    }
+    if (!checkAreas.empty() && !transitionCheck && !messageCloseCheck && !pendingSaleCheck) {
+        for (int i = 0; i < 10; i++) {
+            if (CheckByArea()) {
+                checkCounter = 0;
+                break;
+            } else {
+                checkCounter++;
+            }
+        }
+        if (checkLoops > 15) {
+            checkAreas.erase(checkAreas.begin());
+            checkLoops = 0;
+        }
+    }
+    if (savedFrames > 0 && !pendingSaleCheck && !messageCloseCheck) {
+        savedFrames--;
+    }
+}
+
+void CheckTrackerSaleEnd(GetItemEntry giEntry) {
+    if (pendingSaleCheck) {
+        pendingSaleCheck = false;
+    }
+}
+
+void CheckTrackerItemReceive(GetItemEntry giEntry) {
+    if (!GameInteractor::IsSaveLoaded() || vector_contains_scene(skipScenes, gPlayState->sceneNum)) {
+        return;
+    }
+    auto scene = static_cast<SceneID>(gPlayState->sceneNum);
+    // Vanilla special item checks
+    if (!gSaveContext.n64ddFlag) {
+        if (giEntry.itemId == ITEM_SHIELD_DEKU) {
+            SetCheckCollected(RC_KF_SHOP_ITEM_3);
+            return;
+        }
+        if (giEntry.itemId == ITEM_KOKIRI_EMERALD) {
```

Comment:
[nit] Was this supposed to be an else if to match the rest of this conditional block?
```suggestion
        } else if (giEntry.itemId == ITEM_KOKIRI_EMERALD) {
```